flaml/automl.py

```diff
@@ -1122,14 +1183,20 @@ def custom_metric(
 
         self._retrain_full = retrain_full and (
             eval_method == 'holdout' and self._state.X_val is None)
-        self._prepare_data(eval_method, split_ratio, n_splits)
+        if self._state.task != 'forecast':
+            self._prepare_data(eval_method, split_ratio, n_splits)
+        else:
+            self._prepare_data(eval_method, split_ratio, n_splits,
+                               period=self._state.fit_kwargs.get('period'))
         self._sample = sample and eval_method != 'cv' and (
             MIN_SAMPLE_TRAIN * SAMPLE_MULTIPLY_FACTOR < self._state.data_size)
         if 'auto' == metric:

```

Comment:
Consider using `None` instead of a string "auto" as default.

Reply:
I will be leaving this the way it is since this is the original code.